core/src/OpenMPTarget/Kokkos_OpenMPTarget_Exec.hpp

```diff
@@ -1181,27 +1181,35 @@ KOKKOS_INLINE_FUNCTION void parallel_scan(
     const Impl::TeamThreadRangeBoundariesStruct<
         iType, Impl::OpenMPTargetExecTeamMember>& loop_bounds,
     const FunctorType& lambda) {
-  using ValueTraits = Kokkos::Impl::FunctorValueTraits<FunctorType, void>;
-  using value_type  = typename ValueTraits::value_type;
-
-  auto scan_val = value_type{};
-
-  // Intra-member scan
-#ifdef KOKKOS_ENABLE_PRAGMA_IVDEP
-#pragma ivdep
-#endif
-  for (iType i = loop_bounds.start; i < loop_bounds.end; ++i) {
-    lambda(i, scan_val, false);
-  }
-
-  // 'scan_val' output is the exclusive prefix sum
-  scan_val = loop_bounds.team.team_scan(scan_val);
-
-#ifdef KOKKOS_ENABLE_PRAGMA_IVDEP
-#pragma ivdep
-#endif
-  for (iType i = loop_bounds.start; i < loop_bounds.end; ++i) {
-    lambda(i, scan_val, true);
+  // Extract value_type from lambda
+  using value_type = typename Kokkos::Impl::FunctorAnalysis<
+      Kokkos::Impl::FunctorPatternInterface::SCAN, void,
+      FunctorType>::value_type;
+
+  const auto start     = loop_bounds.start;
+  const auto end       = loop_bounds.end;
+  auto& member         = loop_bounds.team;
```

Comment:
```diff
--- OpenMPTarget
+++ Cuda
@@ -1,7 +1,7 @@
-template <typename iType, class FunctorType>
+template <typename iType, typename FunctorType>
 KOKKOS_INLINE_FUNCTION void parallel_scan(
-    const Impl::TeamThreadRangeBoundariesStruct<
-        iType, Impl::OpenMPTargetExecTeamMember>& loop_bounds,
+    const Impl::TeamThreadRangeBoundariesStruct<iType, Impl::CudaTeamMember>&
+        loop_bounds,
     const FunctorType& lambda) {
   // Extract value_type from lambda
   using value_type = typename Kokkos::Impl::FunctorAnalysis<
@@ -10,7 +10,7 @@

   const auto start     = loop_bounds.start;
   const auto end       = loop_bounds.end;
-  auto& member         = loop_bounds.team;
+  auto& member         = loop_bounds.member;
   const auto team_size = member.team_size();
   const auto team_rank = member.team_rank();
   const auto nchunk    = (end - start + team_size - 1) / team_size;

Reply:
Is the diff in the definition of `member` intentional?

Reply:
yes, because the specialization of TeamThreadRangeBoundariesStruct is different ...

I promise this compiles, it just failed at runtime because of the missing actual implementaton of team_scan and team_broadcast

Reply:
A comment to this effect would be helpful. Having to diff things like this is definitely not ideal.